src/client/Microsoft.Identity.Client/OAuth2/Throttling/UiRequiredProvider.cs

```diff
@@ -118,6 +119,13 @@ private static bool IsRequestSupported(AuthenticationRequestParameters requestPa
                 sb.Append(crypto.CreateSha256Hash(rt) + ThrottleCommon.KeyDelimiter);
             }
 
+            // check mam enrollment id
+            if (bodyParams.TryGetValue(SilentRequestHelper.MamEnrollmentIdKey, out string mamEnrollmentId) &&
+                !string.IsNullOrEmpty(rt))
```

Comment:
if there's an enrollemnt key, just add it to the hash, there's no need to check if RT is null.